app/src/main/kotlin/com/wire/android/ui/calling/initiating/InitiatingCallViewModel.kt

```diff
@@ -35,39 +32,58 @@ class InitiatingCallViewModel @Inject constructor(
     private val conversationDetails: ObserveConversationDetailsUseCase,
     private val allCalls: GetAllCallsUseCase,
     private val startCall: StartCallUseCase,
-    private val endCall: EndCallUseCase,
     private val callRinger: CallRinger
 ) : ViewModel() {
 
-    var callInitiatedState by mutableStateOf(InitiatingCallState())
-        private set
-
     val conversationId: QualifiedID = savedStateHandle
         .get<String>(EXTRA_CONVERSATION_ID)!!
         .parseIntoQualifiedID()
 
+    val conversationType: Deferred<ConversationType> = retrieveConversationTypeAsync()
```

Comment:
why we want to do it that way ? I see that we are using it initiateCall only right ? you wouldn;t need to make a global variable I think than

Reply:
yeah it's private..I will add that modifier
We need it Deferred as we need that for starting a new call
`
startCall(
            conversationId = conversationId,
            conversationType = conversationType.await()
        )
`